merkle-tree/src/oneshot.rs

```diff
@@ -1,31 +1,216 @@
 use simperby_common::crypto::Hash256;
 use simperby_common::MerkleProof;
+use simperby_common::MerkleTree;
 
 /// A Merkle tree that is created once but never modified.
 ///
 /// This is useful for per-block data such as transaction lists.
 pub struct OneshotMerkleTree {
-    // TODO
+    hash_list: Vec<Hash256>,
 }
 
 impl OneshotMerkleTree {
     pub const EMPTY_HASH: Hash256 = Hash256::zero();
 
-    pub fn create(_data: Vec<Hash256>) -> Self {
-        unimplemented!()
+    /// Creates a new OneshotMerkleTree from the given data.
+    pub fn create(data: Vec<Hash256>) -> Self {
+        OneshotMerkleTree { hash_list: data }
     }
 
     /// Creates a Merkle proof for a given data in the tree.
     ///
     /// Returns `None` if the data is not in the tree.
-    pub fn create_merkle_proof(&self, _key: Hash256) -> Option<MerkleProof> {
-        unimplemented!()
+    ///
+    /// Given a tree [[1, 2, 3], [4, 5], [6]],
+    /// Merkle proof for 2 is [(1, false), (5, true)] and Merkle proof for 3 is [(OnlyChildNode, true), (4, false)].
```

Comment:
Please update the description (no boolean)

Reply:
Fixed.